app/src/main/java/org/wikipedia/notifications/NotificationActivity.kt

```diff
@@ -0,0 +1,643 @@
+        currentContinueStr = null
+        disposables.clear()
+
+        // if we're not checking for unread notifications, then short-circuit straight to fetching them.
+        if (displayArchived) {
+            orContinueNotifications
+            return
+        }
+        disposables.add(ServiceFactory.get(WikiSite(Service.COMMONS_URL)).unreadNotificationWikis
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe({ response: MwQueryResponse ->
+                    val wikiMap = response.query()!!.unreadNotificationWikis()
+                    dbNameMap.clear()
+                    for (key in wikiMap!!.keys) {
+                        if (wikiMap[key]!!.source != null) {
+                            dbNameMap[key] = WikiSite(wikiMap[key]!!.source!!.base)
+                        }
+                    }
+                    orContinueNotifications
+                }) { t: Throwable -> setErrorState(t) })
+    }
+
+    private val orContinueNotifications: Unit
+        get() {
+            binding.notificationsProgressBar.visibility = View.VISIBLE
+            disposables.add(ServiceFactory.get(WikiSite(Service.COMMONS_URL)).getAllNotifications("*", if (displayArchived) "read" else "!read", currentContinueStr)
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe({ response: MwQueryResponse ->
+                        onNotificationsComplete(response.query()!!.notifications()!!.list()!!, !currentContinueStr.isNullOrEmpty())
+                        currentContinueStr = response.query()!!.notifications()!!.getContinue()
+                    }) { t: Throwable -> setErrorState(t) })
+        }
+
+    private fun setErrorState(t: Throwable) {
+        L.e(t)
+        binding.notificationsProgressBar.visibility = View.GONE
+        binding.notificationsRecyclerView.visibility = View.GONE
+        binding.notificationsEmptyContainer.visibility = View.GONE
+        binding.notificationsErrorView.setError(t)
+        binding.notificationsErrorView.visibility = View.VISIBLE
+    }
+
+    private fun setSuccessState() {
+        binding.notificationsProgressBar.visibility = View.GONE
+        binding.notificationsErrorView.visibility = View.GONE
+        binding.notificationsRecyclerView.visibility = View.VISIBLE
+    }
+
+    private fun onNotificationsComplete(notifications: List<Notification>, fromContinuation: Boolean) {
+        setSuccessState()
+        if (!fromContinuation) {
+            notificationList.clear()
+            binding.notificationsRecyclerView.adapter = NotificationItemAdapter()
+        }
+        for (n in notifications) {
+            var exists = false
+            for (nExisting in notificationList) {
+                if (nExisting.id() == n.id()) {
+                    exists = true
+                    break
+                }
+            }
```

Comment:
Can this be simplified to the following?

```
notificationList.any { it.id() == n.id() }
```

Reply:
yep! but probably better to use `none {}` instead of `any {}` here.